src/themes/default/components/core/ProductImage.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <div>
+    <img
+      src="/assets/placeholder.svg"
+      v-show="showPlaceholder"
+      itemprop="image"
+      key="placeholderImage"
+      ref="images"
+      class="mw-100 inline-flex">
+    <img
+      :src="images.loading"
+      :alt="alt"
+      v-if="!lowerQualityImageError || isOnline"
+      v-show="showLowerQualityImage"
+      @load="lowerQualityImageLoaded(true)"
+      @error="lowerQualityImageLoaded(false)"
+      itemprop="image"
+      key="lowQualityImage"
+      ref="images"
+      class="mw-100 inline-flex">
+    <img
+      :src="images.src"
+      :alt="alt"
+      v-if="!highQualityImageError || isOnline"
+      v-show="showHighQualityImage"
+      @load="highQualityImageLoaded(true)"
+      @error="highQualityImageLoaded(false)"
+      itemprop="image"
+      key="highQualityImage"
+      ref="images"
+      class="mw-100 inline-flex">
+  </div>
+</template>
+<script>
+import { onlineHelper } from '@vue-storefront/core/helpers'
+export default {
+  props: {
+    images: {
+      type: Object,
+      required: true
+    },
+    alt: {
+      type: String,
+      required: true
+    },
+    hidden: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      lowerQualityImageLoad: false,
+      lowerQualityImageError: true,
+      highQualityImageLoad: false,
+      highQualityImageError: true
+    }
+  },
+  computed: {
+    showPlaceholder () {
+      return !this.showLowerQualityImage && !this.showHighQualityImage
+    },
+    showLowerQualityImage () {
+      return !!this.lowerQualityImageLoad && !this.highQualityImageLoad && !this.hidden
```

Comment:
Isn’t the “!!” a typo?

Reply:
In this case "!!" is not necessary becouse I have everywhere have Boolen and no needed conver variable to Boolen. Remove it?

Reply:
Yes Please. Please do also check the offline mode after this changes - should work without any issues as on demo

Reply:
Done. I find bug on initial valu but fixed it.